tarodb/asset_minting.go

```diff
@@ -231,6 +234,11 @@ func (a *AssetMintingStore) CommitMintingBatch(ctx context.Context,
 
 	var writeTxOpts AssetStoreTxOptions
 	return a.db.ExecTx(ctx, &writeTxOpts, func(q PendingAssetStore) error {
+		if err := validateCandidateSeedlings(ctx, q, newBatch.Seedlings); err != nil {
```

Comment:
What about just relying on the failed constraint during insertion? This way, we don't need to fetch all the seedlings (in the future, will be very minting batch ever created). 

I think there should be a specific error we can use `errors.Is` on, here's a fragment that triggers the error case directly: https://github.com/lightninglabs/taro/blob/77b69191fc60c57135c70e25b338d5d401bfbb2b/tarodb/addrs_test.go#L403-L404

Reply:
I couldn't find a specific sqlite error type (sentinel?) at [this point](https://github.com/lightninglabs/taro/pull/190/files#diff-53cb63ce68ab3788f0991694aa62d7c2d69b019fff9ac916d6d26cea699a8d82R270). The error struct consists of a generic message field and an error code. But I don't think the error code relates to the `asset_name` field in particular.

In the latest commit I'm just inspecting the error message string and converting into a non-sql error message. Is there a better way? Should I cast the error at the `CommitMintingBatch` call site instead?

Reply:
So there's a `sqlite.Error` value which is used to return all errors: https://pkg.go.dev/modernc.org/sqlite#Error

This has a `Code` method, I printed out the error we get in that test linked above, and looks to be `275` which matches up with this: https://gitlab.com/cznic/sqlite/-/blob/v1.19.4/internal/mptest/main_darwin_arm64.go#L490

So I think we can do something like this (to check for constraint violations and return a nicer error):
```go
var sqlError *sqlite.Error
if errors.As(err, &sqlError) && sqlErr.Code() == sqlite.SQLITE_CONSTRAINT_CHECK{
    return ErrDuplicateSeedlingName
}
```

Reply:
That's strange, my computer gives me error code [`SQLITE_CONSTRAINT_UNIQUE = 2067`](https://gitlab.com/cznic/sqlite/-/blob/v1.19.4/internal/mptest/main_darwin_amd64.go#L455). I've enhanced the test to check for the specific error type which is only used if the sqlite error code is `SQLITE_CONSTRAINT_UNIQUE`.

Thanks for the code snippet, I'll go with what you've suggested.